### DISCUSSION TITLE
cheatsheets/macos/

### DISCUSSION THREAD

**Discussion by giscus** [shane0/cheatsheets#26] [General]
# cheatsheets/macos/

https://shanenull.com/cheatsheets/macos/

**Comment by shane0** [comment]
# apps

- https://github.com/RhetTbull/textinator/releases

**Reply by shane0** [reply]
```sh
brew install newsboat
```

**Reply by shane0** [reply]
- https://obsidian.md/

**Comment by shane0** [comment]
# vscode workflow/

- update broke path `code` is missing
- vscode command pallete
- install fails

## workaround

- uninstall asks for pwd
- install asks for pwd

```text
cat << EOF >> ~/.bash_profile
# Add Visual Studio Code (code)
export PATH="$PATH:/Applications/Visual Studio [Code.app/Contents/Resources/app/bin](https://code.app/Contents/Resources/app/bin)"
EOF
```

## testing mkdoc grid

- need version nos

```sh
mkdocs --version
pip show mkdocs-material
```

**Reply by shane0** [reply]
## make finder not suck

- one problem on mac is they hide the full path from you making automation a big pita
- tech lock in
- removing user control and choice and option
- user hostility
- this maze has an exit it is called frame.work laptops running linux


```sh
defaults write com.apple.finder ShowPathbar YES
```

**Reply by shane0** [reply]
## ranger

- copy path
- mark folders and jump
- bulk rename

**Reply by shane0** [reply]
## Top 20 User-Friendly Defaults to Set

**1. Enable Dark Mode:**
   * Go to **System Preferences** > **Appearance**.
   * Select **Dark**.

**2. Adjust Cursor Size:**
   * Go to **System Preferences** > **Accessibility** > **Display**.
   * Adjust the **Cursor size** slider.

**3. Customize Dock Appearance:**
   * Right-click on the Dock and select **Preferences**.
   * Customize the Dock's appearance, size, and behavior.

**4. Enable Mission Control:**
   * Go to **System Preferences** > **Mission Control**.
   * Enable **Mission Control** and customize its settings.

**5. Configure Keyboard Shortcuts:**
   * Go to **System Preferences** > **Keyboard**.
   * Customize keyboard shortcuts for various tasks.

**6. Adjust Trackpad Sensitivity:**
   * Go to **System Preferences** > **Trackpad**.
   * Adjust the **Tracking speed** and **Scroll & Zoom** settings.

**7. Enable Natural Scrolling:**
   * Go to **System Preferences** > **Trackpad**.
   * Check **Scroll & Zoom** and enable **Natural scrolling**.

**8. Set Default Application Associations:**
   * Go to **System Preferences** > **General**.
   * Customize the default applications for different file types.

**9. Enable Accessibility Features:**
   * Go to **System Preferences** > **Accessibility**.
   * Explore and enable accessibility features like VoiceOver, Zoom, or Dictation.

**10. Set a Screen Saver:**
   * Go to **System Preferences** > **Desktop & Dock**.
   * Choose a screen saver and customize its settings.

**11. Enable iCloud Drive:**
   * Go to **System Preferences** > **Apple ID** > **iCloud**.
   * Enable **iCloud Drive** and choose your preferences.

**12. Configure Notifications:**
   * Go to **System Preferences** > **Notifications**.
   * Customize notification settings for apps.

**13. Adjust Display Settings:**
   * Go to **System Preferences** > **Displays**.
   * Adjust brightness, resolution, and color profiles.

**14. Set a Password Autofill Service:**
   * Go to **System Preferences** > **Passwords & Accounts**.
   * Choose your preferred password autofill service.

**15. Enable Automatic Updates:**
   * Go to **System Preferences** > **Software Update**.
   * Enable **Automatic updates**.

**16. Configure Energy Saver:**
   * Go to **System Preferences** > **Energy Saver**.
   * Adjust energy-saving settings.

**17. Customize Spotlight Search:**
   * Go to **System Preferences** > **Spotlight**.
   * Customize Spotlight's search behavior and results.

**18. Configure Siri:**
   * Go to **System Preferences** > **Siri**.
   * Enable Siri and customize its settings.

**19. Set a Time Machine Backup:**
   * Go to **System Preferences** > **Time Machine**.
   * Configure Time Machine to automatically back up your data.

**20. Manage Apple ID Settings:**
   * Go to **System Preferences** > **Apple ID**.
   * Manage your Apple ID information, iCloud settings, and more.

**Reply by shane0** [reply]
```sh
# Set default application associations:
defaults write com.apple.finder AppleDefaultApplications -dict-of-string "application/pdf" "Preview" "public.image" "Preview"

# Enable or disable automatic updates:
defaults write com.apple.SoftwareUpdate AutomaticCheckEnabled 0  # disable
defaults write com.apple.SoftwareUpdate AutomaticCheckEnabled 1  # enable

# Adjust Dock appearance:
defaults write com.apple.dock minimize-to-application -bool YES  # minimize to application
defaults write com.apple.dock minimize-to-application -bool NO  # minimize to original position
defaults write com.apple.dock autohide -bool YES  # autohide
defaults write com.apple.dock autohide -bool NO  # show always

# Configure Mission Control:
defaults write com.apple.dock expose-delay -float 0.2  # adjust exposure delay
defaults write com.apple.dock expose-animation-duration -float 0.1  # adjust animation duration

# Adjust trackpad sensitivity:
defaults write -g ApplePointerSpeed -float 1.5  # adjust tracking speed
defaults write -g ApplePressAndHoldEnabled -bool NO  # disable tap to click

# Enable or disable natural scrolling:
defaults write com.apple.trackpad ScrollInverted -bool YES  # enable
defaults write com.apple.trackpad ScrollInverted -bool NO  # disable

# Enable or disable Siri:
defaults write com.apple.Siri.EnableSiri -bool YES  # enable
defaults write com.apple.Siri.EnableSiri -bool NO  # disable

# Adjust screen saver settings:
defaults write com.apple.screensaver.fadescreen Delay 30  # set fade delay to 30 seconds
defaults write com.apple.screensaver.fadescreen IdleTime 600  # set idle time before fade to 10 minutes

# Configure notifications:
defaults write com.apple.notificationcenter.notificationcenterui ShowBadgeCount -bool NO  # hide badge count
defaults write com.apple.notificationcenter.notificationcenterui ShowNotificationPreviews -bool NO  # hide notification previews

# Adjust display settings:
defaults write com.apple.AppleMulticolorDisplay Brightness 0.8  # set brightness to 80%
defaults write com.apple.AppleMulticolorDisplay Gamma 2.2  # set gamma to 2.2

# Configure iCloud Drive:
defaults write com.apple.finder.iCloud.DisableDesktopAndDocumentsSync -bool YES  # disable Desktop and Documents sync
defaults write com.apple.finder.iCloud.DisableDesktopAndDocumentsSync -bool NO  # enable Desktop and Documents sync

# Configure energy saver:
defaults write com.apple.PowerManagement autosleep-delay 600  # set sleep delay to 10 minutes
defaults write com.apple.PowerManagement.controlPanel activeDisplayTimeout 300  # set display timeout to 5 minutes

# Customize Spotlight Search:
defaults write com.apple.spotlight.plist SuppressSpotlight -bool YES  # disable Spotlight
defaults write com.apple.spotlight.plist SuppressSpotlight -bool NO  # enable Spotlight

# Configure Time Machine:
defaults write com.apple.TimeMachine.Enable -bool NO  # disable Time Machine
defaults write com.apple.TimeMachine.Enable -bool YES  # enable Time Machine

# Manage Apple ID settings:
defaults write com.apple.loginwindow LoginHook /path/to/your/script  # set a login hook
defaults write com.apple.loginwindow LogoutHook /path/to/your/script  # set a logout hook
```

**Comment by shane0** [comment]
# brew

```sh
brew install ranger
```

**Reply by shane0** [reply]
# sd cards

- sync

```sh
`diff /Users/<username>/Downloads/lacie-backups /Volumes/400g/lacie-backups` 
```

**Comment by shane0** [comment]
# macos being anti developer

- aka reasons to go back to linux
- switched shell to zsh
- reboot or updates break the path to `code` recently

**Reply by shane0** [reply]
# macos uninstalls code shell command

- https://www.reddit.com/r/vscode/comments/qsikaj/the_code_shell_command_keeps_uninstalling_from_my/

**Comment by shane0** [comment]
# show hidden

To show hidden files on macOS, follow these steps:

### Method 1: Keyboard Shortcut (Finder)
1. Open a Finder window.
2. Press **Command (⌘) + Shift + . (Period)**.  
   - Hidden files will appear as slightly grayed-out icons.
   - Press the same shortcut again to hide them.

### Method 2: Terminal Command
1. Open the **Terminal** app.
2. Run the following command to show hidden files:
   ```bash
   defaults write com.apple.finder AppleShowAllFiles -bool true
   ```
3. Restart Finder for the changes to take effect:
   ```bash
   killall Finder
   ```
4. To hide the files again, run:
   ```bash
   defaults write com.apple.finder AppleShowAllFiles -bool false
   ```
5. Restart Finder again:
   ```bash
   killall Finder
   ```

Choose the method that works best for your needs!
